Manually insert ToC

backflip opened the pull request:
With this change, the table of contents is no longer inserted automatically but rather via manually adding `[[toc]]` placeholder wherever needed.

UPDATE: See below, the default placeholder is now `[[_TOC_]]`.

backflip committed:
feat: change ToC insertion to manual mode

backflip committed:
fix: use same slug method for toc and anchors

backflip commented:
@renestalder, is the default `[[toc]]` placeholder a good solution or do you know of a better one?

christophdubach committed:
Merge master’s css into feature/toc-manual_css

christophdubach committed:
feat: optimize toc spacings

christophdubach commented:
Works as expected, so for me it’s ready to get merged.
But there’s still the question if the placeholder `[[toc]]` is okay ...

backflip commented:
@renestalder, unless you intervene we are going to merge this. Changing the placeholder should be easy at any point in time.

renestalder commented:
@backflip Sorry, I missed that PR.

Interesting enough, I thought there would be some standard around the table of content syntax in Markdown, but yeah, there isn't and my research only resulted in one markdown flavoured addition, the one from GitLab.  [GitLab uses `[[_TOC_]]` or `[TOC]`](https://docs.gitlab.com/ee/user/markdown.html#table-of-contents).

Most other tools insert table of contents automatically and mark the start and end of them with code comments.

So, well, actually I have no points to bring up any more regarding the exact placeholder to use.

Should we add something to the readme/contributor guidelines regarding this "non-standard" placeholder?

backflip committed:
feat: change toc placeholder

backflip committed:
docs: mention toc in contribution guidelines

backflip commented:
@renestalder, thanks a bunch! I have replaced the `[[toc]]` with `[[_TOC_]]` as the plugin already supports that out of the box, too. And as Gitlab is using and documenting it, I'd say this is the closest we get to a "standard".

I have added a sentence to `CONTRIBUTING.md`, too, thanks for the hint!
